connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStore.java

```diff
@@ -279,10 +280,33 @@ public Future<Void> set(Map<ByteBuffer, ByteBuffer> values, Callback<Void> callb
             throw new IllegalStateException("At least one non-null offset store must be provided");
         }
 
+        boolean containsTombstones = values.entrySet()
+                .stream()
+                .anyMatch(offset -> offset.getValue() == null);
```

Comment:
nit: can be simplified
```suggestion
        boolean containsTombstones = values.containsValue(null);
```

Reply:
We should do this only if `connectorStore.isPresent()` else we will always end up scanning the map for tombstones.

Reply:
@ashwinpankaj , I am thinking it should be ok to scan the map because it's not going to be huge. Moreover, the flag `containsTombstones` is used when writing to primary store as well to check if secondary store writes have already happened. WDYT?